components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.tsx

```diff
@@ -243,6 +235,19 @@ export default class PermissionsTree extends React.PureComponent<Props, State> {
         if (!this.props.customGroupsEnabled) {
             customGroupsGroup?.permissions.pop();
         }
+
+        if (license?.SkuShortName === LicenseSkus.Enterprise) {
+            this.groups[3].permissions.push(Permissions.PLAYBOOK_PUBLIC_MAKE_PRIVATE);
+            this.groups.splice(4, 0, {
```

Comment:
Opting to splice into the groups structure vs. filter to keep things consistent with the other modifications being made in place.

Reply:
I think https://github.com/mattermost/mattermost-webapp/compare/declarative-permissions-visibility would be a better approach because:
a) it keeps the information about visibility in the same place as the rest of the section's configuration, and
b) it avoids accessing the object by index when mutating it

Reply:
Thanks for that original PR, @mkraft! The challenge I faced was the need not only to filter the groups, but also specific permissions within the group. Should I try filtering on groups and keeping the modify-in-place code that matches the way the other permissions are managed?